src/odemis/driver/test/orsay_test.py

```diff
@@ -0,0 +1,652 @@
+            self.fail(e)
+        self.assertEqual(len(oserver.children.value), 4)
+
+        oserver.terminate()
+
+    def test_wrong_ip(self):
+        """
+        Tests that an HwError is raised when a wrong ip address is entered
+        """
+        with self.assertRaises(HwError):
+            orsay.OrsayComponent(name="Orsay", role="orsay", host="192.168.1.1", children="")
+
+
+class TestOrsay(unittest.TestCase):
+    """
+    Tests to run on the main Orsay component
+    """
+
+    oserver = None
+
+    @classmethod
+    def setUpClass(cls):
+        """
+        Setup the Orsay client
+        """
+        if TEST_NOHW == 1:
+            raise unittest.SkipTest("TEST_NOHW is set. No server to contact.")
+        try:
+            cls.oserver = orsay.OrsayComponent(**CONFIG_ORSAY)
+        except Exception as e:
+            raise HwError(str(e))
+
+        cls.datamodel = cls.oserver.datamodel
+
+        for child in cls.oserver.children.value:
+            if child.name == CONFIG_PSUS["name"]:
+                cls.psus = child
+            elif child.name == CONFIG_PRESSURE["name"]:
+                cls.pressure = child
+            elif child.name == CONFIG_PSYS["name"]:
+                cls.psys = child
+            elif child.name == CONFIG_UPS["name"]:
+                cls.ups = child
+
+    @classmethod
+    def tearDownClass(cls):
+        """
+        Terminate the Orsay client
+        """
+        cls.oserver.terminate()
+
+    def test_updateProcessInfo(self):
+        """
+        Check that the processInfo VA is updated properly and an exception is raised when the wrong parameter is passed
+        """
+        with self.assertRaises(ValueError):
+            self.oserver._updateProcessInfo(self.datamodel.HybridPlatform.Cancel)
+        if not TEST_NOHW == "sim":
+            self.skipTest("TEST_NOHW is not set to sim, cannot force data on Actual parameters of Orsay server "
+                          "outside of simulation.")
+        test_string = "Some process information"
+        self.datamodel.HybridPlatform.ProcessInfo.Actual = test_string
+        self.assertEqual(self.oserver.processInfo.value, test_string)
+        self.datamodel.HybridPlatform.ProcessInfo.Actual = ""
+
+    def test_reconnection(self):
+        """
+        Checks that after reconnection things still work
+        """
+        self.oserver._device.HttpConnection.close()  # close the connection
+        self.oserver._device.MessageConnection.Connection.close()
+        self.oserver._device.DataConnection.Connection.close()
+        self.oserver._device.MessageConnection.dataConnection.Connection.close()
+        sleep(1)
+        while not self.oserver.state.value == model.ST_RUNNING:
+            sleep(2)  # wait for the reconnection
+
+        # perform some test to check writing and reading still works
+        self.psus._valve.Target = 1
+        sleep(1)
+        self.assertTrue(self.psus.power.value)
+
+        self.psus._valve.Target = 2
```

Comment:
For all these valve.Target = 1/2 , could you replace them by VALVE_OPEN and VALVE_CLOSE? Otherwise, it's very hard to read the test cases.